tests/test_resource_warning.py

```diff
@@ -0,0 +1,68 @@
+import asyncio
+import time
+import typing
+import warnings
+from datetime import datetime
+
+import pytest
+
+import aiohttp
+
+last_warning: typing.Optional[str] = None
+
+
+def print_warnings(message, category, filename, lineno, file=None, line=None):
+    global last_warning
+    text = f'{datetime.now()} | ' \
+           f'{filename}:{lineno} {category.__name__:s}:{message}'
+    print(text)
+    # These are warnings in the 4.0a and these prevent execution of the test
+    if "RuntimeWarning:coroutine 'noop' was never awaited" in text:
+        return None
+    last_warning = text
+
+
+@pytest.fixture()
+def warning_setup(monkeypatch):
+    # setup warning
+    monkeypatch.setattr(warnings, 'showwarning', print_warnings)
+    warnings.simplefilter('default')
+
+
+async def _resource_warning():
+    timeout = 3 * 60
+
+    client = aiohttp.ClientSession()
+
+    request_interval = 30
+    request_increase = 15
+    next_request = 0
+
+    end = time.time() + timeout
+    while last_warning is None:
+        now = time.time()
+        if now > end:
+            break
+
+        if now > next_request:
+            async with client.get('https://heise.de') as resp:
+                status = resp.status
+                await resp.text()
+
+            request_interval += request_increase
+            next_request = now + request_interval
+            print(f'{datetime.now()} | '
+                  f'Status: {status}, next request in {request_interval}s')
+
+        await asyncio.sleep(5)
+
+    await client.close()
+    assert last_warning is None, last_warning
+
+
+def test_resource_warning(warning_setup):
+    # the future has to be run like this, with the pytest async runner
+    # the warnings appear only after the test
+    loop = asyncio.new_event_loop()
+    loop.run_until_complete(_resource_warning())
+    loop.close()
```

Comment:
How about
```suggestion
def test_resource_warning(recwarn, warning_setup):
    # the future has to be run like this, with the pytest async runner
    # the warnings appear only after the test
    loop = asyncio.new_event_loop()
    loop.run_until_complete(_resource_warning())
    loop.close()
    assert not str(recwarn.pop(RuntimeWarning).message).startswith(
        'unclosed transport '
        '<asyncio.sslproto._SSLProtocolTransport object at ',
    )
```
(but using `pytest.warns()` CM` would also work)

Reply:
This works
```python
    for warn in recwarn:
        assert 'unclosed transport' not in str(warn.message)
```
but provides not the fast exit on the first warning keeping the test running for some unnecessary time.

Reply:
What if you pass `recwarn` object as a param to `_resource_warning()` and then use it inside on each iteration? Will it work via closure? (as in recording the outer warnings that the loop emits while having access to them from the inside)
```python
    loop.run_until_complete(_resource_warning(recwarn))
```
this way you could terminate the loop early.